src/components/flyout/flyout.tsx

```diff
@@ -263,35 +254,33 @@ export const EuiFlyout = forwardRef(
     };
 
     let newStyle;
-    let widthClassName;
-    let sizeClassName;
 
     // Setting max-width
-    if (maxWidth === true) {
-      widthClassName = 'euiFlyout--maxWidth-default';
-    } else if (maxWidth !== false) {
+    if (maxWidth !== false) {
       const value = typeof maxWidth === 'number' ? `${maxWidth}px` : maxWidth;
       newStyle = { ...style, maxWidth: value };
     }
```

Comment:
Having looked closer at the maxWidth CSS this conditional is also now incorrect according to the props description

```suggestion
    if (maxWidth !== true) {
      newStyle = { ...style, maxWidth: maxWidth === false ? 'none' : maxWidth };
    }
```

for context on my removal of line 260 - there's no need to convert maxWidth to a `px` string, React intelligently converts numbers to px strings

Reply:
I've been playing around with this one, and I think it may be helpful for us to restore the `widthClassName = 'euiFlyout--maxWidth-default';`.

Per [this comment](https://github.com/elastic/eui/pull/6213#discussion_r965239408). I created a function to DRY out the repetitive styling of the Flyout sizing.  It was recommended in that comment that we use something like this 👇🏾  to set the maxWidth for the Flyout, however, we should only be setting this max width when the `maxWidth` prop is true.
```
${logicalCSS('max-width', `${flyoutSizes.m.max}px`)}
```

Instead of creating another entry to the Flyout styles map (and another DRY helper), I could just target the `euiFlyout--maxWidth-default` class name directly and apply the styles when necessary. 

Would this be favorable? 

```
[class*="euiFlyout--maxWidth-default"] { ${logicalCSS('max-width', flyoutSizes[size].max)} }
```

Reply:
In this scenario I don't think any extra code cruft added by `maxWidth` logic is worth adding a modifier class and CSS nesting. This should be relatively achievable by JS alone.

I'd go with one of two options:

1. Leave `${logicalCSS('max-width', `${flyoutSizes.m.max}px`)}` in as-is and override `maxWidth` to none if maxWidth is false. This is what my code above does in inline `style`, but if you prefer to avoid the style tag (a reasonable preference), then you can add another conditional style to the array, e.g. `[ maxWidth ===  false && styles.noMaxWidth]` which should set `max-width: none;` and come after the flyout sizing CSS (thus overriding it).

2. Pass in a custom `defaultMaxWidth` variable to `euiFlyoutStyles()` (note: this should only pass the true/false boolean for using the default max width vs `none`, and should *not* pass the custom maxWidth strings - those should remain as inline styles). This is much a pretty decent tradeoff in terms of readability vs Emotion generation, as since this a single boolean flag it will only generate 2 classes on the fly vs potentially infinite.

Let me know if you have any questions, super happy to clarify or pair if needed!